src/plugins/navigation_embeddable/public/components/dashboard_link/dashboard_link_component.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import useAsync from 'react-use/lib/useAsync';
+
+import { EuiButtonEmpty } from '@elastic/eui';
+import { DashboardContainer } from '@kbn/dashboard-plugin/public/dashboard_container';
+
+import {
+  DASHBOARD_LINK_TYPE,
+  NavigationEmbeddableLink,
+  NavigationLinkInfo,
+} from '../../embeddable/types';
+import { fetchDashboard } from './dashboard_link_tools';
+import { useNavigationEmbeddable } from '../../embeddable/navigation_embeddable';
+
+export const DashboardLinkComponent = ({ link }: { link: NavigationEmbeddableLink }) => {
+  const navEmbeddable = useNavigationEmbeddable();
+
+  const dashboardContainer = navEmbeddable.parent as DashboardContainer;
+  const parentDashboardTitle = dashboardContainer.select((state) => state.explicitInput.title);
+  const parentDashboardId = dashboardContainer.select((state) => state.componentState.lastSavedId);
+
+  const { loading: loadingDestinationDashboard, value: destinationDashboard } =
+    useAsync(async () => {
+      return await fetchDashboard(link.destination);
+    }, [link, parentDashboardId]);
```

Comment:
Consider the following scenario:

1. You add a link to your current dashboard to a navigation embeddable
2. You rename your current dashboard **without** saving as/cloning, so the `parentDashboardId` stays the same
3. The displayed text for the link added in step one **should** update to the new title
4. Now, you `Save as` your dashboard with a **different** title from step 2 - so, `parentDashboardId` has now changed
5. As expected, the link from step 1 no longer points to your "current" dashboard, so it is now clickable - also, because the `parentDashboardId`, we re-fetch the `destinationDashboard` so that we can get the updated title from step 3, since `parentDashboardTitle` is no longer representing it.

That is why I added `parentDashboardId` to the dependencies here